tests/ios/app/BUILD.bazel

```diff
@@ -112,6 +112,14 @@ ios_application(
     ],
 )
 
+ios_unit_test(
+    name = "TestAppWithProvisioningProfile",
+    srcs = ["empty_tests.m"],
+    minimum_os_version = "10.0",
+    provisioning_profile = "@build_bazel_rules_ios//tests/ios:integration-test.mobileprovision",
```

Comment:
We can think of more elaborate test for the "on device" use case, just adding a way to catch some possible failure modes when the attr is set

I believe this use case as whole is missing more test cases and could receive some love but not planning to tackle it in this PR

Reply:
What'd be cool is to add some signing tests with an ad-hoc identity - we might end up needing to bake that into the repo. Longer term we can probably have some end to end integration test with the Xcode GUI stuff as well 👍